File: elixir/line-up/lib/line_up.ex
defmodule LineUp do
  @doc """
  Formats a full ticket sentence for the given name and number, including
  the person's name, the ordinal form of the number, and fixed descriptive text.
  """
  @spec format(name :: String.t(), number :: pos_integer()) :: String.t()
  def format(name, number) do
    numstr = to_string(number)
    ordinal_suffix =
      cond do
        String.ends_with?(numstr, "1") and not String.ends_with?(numstr, "11") -> "st"
        String.ends_with?(numstr, "2") and not String.ends_with?(numstr, "12") -> "nd"
        String.ends_with?(numstr, "3") and not String.ends_with?(numstr, "13") -> "rd"
        true -> "th"
      end
    "#{name}, you are the #{number}#{ordinal_suffix} customer we serve today. Thank you!"
  end
end

File: python/line-up/line_up.py
"""Produce a sentence using name, number and the number's ordinal"""

def line_up(name, number):
    ordinal = (
        'st' if number % 10 == 1 and number % 100 != 11 else
        'nd' if number % 10 == 2 and number % 100 != 12 else
        'rd' if number % 10 == 3 and number % 100 != 13 else
        'th'
    )
    return f'{name}, you are the {number}{ordinal} customer we serve today. Thank you!'
